patch_recurrence/steps.py

```diff
@@ -88,12 +119,24 @@ def generate_pairs(imgs, patches, constants):
     index_database = []
     length_database = []
     for k in range(scaled_imgs):
+        a = [patch.norm_patch for patch in patches[k] if 7 <= patch.bucket <= 9]
```

Comment:
this part is hard to read, what exactly are you trying to do here? We should clean up the parts with `x` and `t`